tests/cli/test_compose_tar_kickstart.sh

```diff
@@ -0,0 +1,158 @@
+
+        fallocate -l 5G disk.img
+    rlPhaseEnd
+
+    rlPhaseStartTest "compose start"
+        rlAssertEquals "SELinux operates in enforcing mode" "$(getenforce)" "Enforcing"
+        uuid=$($CLI compose start $bp_name tar)
+        rlAssertEquals "exit code should be zero" $? 0
+
+        uuid=$(echo $uuid | cut -f 2 -d' ')
+    rlPhaseEnd
+
+    rlPhaseStartTest "compose finished"
+        if [ -n "$uuid" ]; then
+            until $CLI compose info $uuid | grep 'FINISHED\|FAILED'; do
+                sleep 10
+                rlLogInfo "Waiting for compose to finish ..."
+            done;
+        else
+            rlFail "Compose uuid is empty!"
+        fi
+
+        rlRun -t -c "$CLI compose image $uuid"
+        image="$uuid-root.tar.xz"
+    rlPhaseEnd
+
+    rlPhaseStartTest "Install tar image using kickstart liveimg command"
+        cat > $ks_path << __EOF__
+lang en_US.UTF-8
+timezone America/New_York
+keyboard us
+rootpw --lock
+sshkey --username root "$pub_key"
+bootloader --location=mbr
+zerombr
+clearpart --initlabel --all
+autopart
+# reboot is used together with --no-reboot qemu-kvm parameter for the qemu-kvm
+# process to exit after the installation is complete
+reboot
+
+liveimg --url http://10.0.2.2/root.tar.xz
+
+%post --log=/dev/console
+echo -e "curl http://10.0.2.2/image-$uuid-booted\n" >> /etc/rc.d/rc.local
+chmod +x /etc/rc.d/rc.local
+curl http://10.0.2.2/image-$uuid-installed
+%end
+__EOF__
+        mv $image $image_path
+        restorecon $image_path
+        $QEMU_BIN -m 2048 -drive file=disk.img,format=raw -nographic -kernel vmlinuz -initrd initrd.img \
+            -append "inst.ks=http://10.0.2.2/ks-tar.cfg inst.stage2=$baseurl console=ttyS0" --no-reboot &
+
+        timeout=60 # wait up to 60 minutes
+        t=0
+        while ((t < timeout)) && ! grep -q "image-$uuid-installed" /var/log/httpd/access_log; do
+            rlLogInfo "Waiting for the tar image to be installed."
+            sleep 1m
+            ((t += 1))
+        done
+        rlRun -t -c "grep image-$uuid-installed /var/log/httpd/access_log" 0 "Installation of the image finished."
+
+        # rather check explicitly that the installation finished than wait unnecessarily if the installation failed
+        if grep -q "image-$uuid-installed" /var/log/httpd/access_log; then
+            # wait for the installation process to complete and boot the installed system
+            sleep 1m
+            $QEMU_BIN -m 2048 -drive file=disk.img,format=raw -nographic -net user,id=nic0,hostfwd=tcp::2222-:22 \
+                -net nic &
+            t=0
+            timeout=10
+            while ((t < timeout)) && ! grep -q "image-$uuid-booted" /var/log/httpd/access_log; do
+                rlLogInfo "Waiting for the installed system to boot."
+                sleep 1m
+                ((t += 1))
+            done
+            rlRun -t -c "grep image-$uuid-booted /var/log/httpd/access_log" 0 "Installed system booted."
```

Comment:
This will become obsolete with the commit from Brian.

Reply:
> > I saw a comment about IP addresses, and can't tell if it was answered. You can set the subnet to use by passing in net=192.168.103.0/24 as one of the netdev options.
> 
> @jikortus what about this comment? where does the `10.0.2.2` IP address come from ?

Sorry that I missed this comment. 10.0.2.2 is the default host IP address used by the internal qemu-kvm DHCP server (and FTR, 10.0.2.15 is the first guest address in the range), as I found out in the documentation. Of course this can be changed to any custom address range, but I think the defaults should work well for us.

Reply:
OK, if the docs say this is the default it's fine.